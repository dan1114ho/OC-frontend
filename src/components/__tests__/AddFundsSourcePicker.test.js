import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { MockAddFundsSourcePicker, MockAddFundsSourcePickerForUser } from '../AddFundsSourcePicker';

describe('AddFundsSourcePicker component', () => {
  const defaultProps = {
    collective: {},
    data: {},
    host: {},
    onChange: () => {},
    paymentMethod: {},
  };

  it('renders default options', () => {
    const tree = renderer.create(<MockAddFundsSourcePicker {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders loading state', () => {
    const props = {
      ...defaultProps,
      data: {
        loading: true,
      },
    };
    const tree = renderer.create(<MockAddFundsSourcePicker {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders host name as first option', () => {
    const props = {
      ...defaultProps,
      host: {
        id: 'example-id',
        name: 'Example Host',
      },
    };

    const tree = renderer.create(<MockAddFundsSourcePicker {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders fromCollectives by type in optgroup', () => {
    const props = {
      ...defaultProps,
      data: {
        PaymentMethod: {
          fromCollectives: {
            collectives: [
              {
                id: 'collective-id',
                name: 'Example Collective',
                type: 'COLLECTIVE',
              },
              {
                id: 'organization-id',
                name: 'Example Organzation',
                type: 'ORGANIZATION',
              },
              {
                id: 'user-id',
                name: 'Example Person',
                type: 'USER',
              },
            ],
          },
        },
      },
    };

    const tree = renderer.create(<MockAddFundsSourcePicker {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onChange prop function when selection changes', () => {
    const onChange = jest.fn();
    const props = {
      ...defaultProps,
      onChange,
    };

    const component = mount(<MockAddFundsSourcePicker {...props} />);

    component.find('select').simulate('change', { target: { value: 'test' } });

    expect(onChange).toHaveBeenCalledWith('test');
  });
});

describe('AddFundsSourcePickerForUser component', () => {
  const defaultProps = {
    LoggedInUser: {
      hostsUserIsAdminOf: () => [],
    },
    onChange: () => {},
  };

  it('renders default options', () => {
    const tree = renderer.create(<MockAddFundsSourcePickerForUser {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render host options', () => {
    const props = {
      ...defaultProps,
      LoggedInUser: {
        hostsUserIsAdminOf: () => [
          {
            id: 'host-id',
            name: 'Example Host',
          },
        ],
      },
    };

    const tree = renderer.create(<MockAddFundsSourcePickerForUser {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
