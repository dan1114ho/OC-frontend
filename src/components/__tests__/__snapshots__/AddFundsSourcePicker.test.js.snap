// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFundsSourcePicker component renders default options 1`] = `
<select
  className="form-control"
  id="sourcePicker"
  name="template"
  onChange={[Function]}
  placeholder="select"
>
  <option>
    Host ()
  </option>
  <optgroup
    label="OTHER"
  >
    <option
      value="other"
    >
      other (please specify)
    </option>
  </optgroup>
</select>
`;

exports[`AddFundsSourcePicker component renders fromCollectives by type in optgroup 1`] = `
<select
  className="form-control"
  id="sourcePicker"
  name="template"
  onChange={[Function]}
  placeholder="select"
>
  <option>
    Host ()
  </option>
  <optgroup
    label="COLLECTIVE"
  >
    <option
      value="collective-id"
    >
      Example Collective
    </option>
  </optgroup>
  <optgroup
    label="ORGANIZATION"
  >
    <option
      value="organization-id"
    >
      Example Organzation
    </option>
  </optgroup>
  <optgroup
    label="PEOPLE"
  >
    <option
      value="user-id"
    >
      Example Person
    </option>
  </optgroup>
  <optgroup
    label="OTHER"
  >
    <option
      value="other"
    >
      other (please specify)
    </option>
  </optgroup>
</select>
`;

exports[`AddFundsSourcePicker component renders host name as first option 1`] = `
<select
  className="form-control"
  id="sourcePicker"
  name="template"
  onChange={[Function]}
  placeholder="select"
>
  <option
    value="example-id"
  >
    Host (Example Host)
  </option>
  <optgroup
    label="OTHER"
  >
    <option
      value="other"
    >
      other (please specify)
    </option>
  </optgroup>
</select>
`;

exports[`AddFundsSourcePicker component renders loading state 1`] = `<div />`;

exports[`AddFundsSourcePickerForUser component render host options 1`] = `
<div>
  <select
    className="form-control"
    id="sourcePicker"
    name="template"
    onChange={[Function]}
    placeholder="select"
  >
    <option
      value=""
    />
    <option
      value="host-id"
    >
      Example Host
    </option>
  </select>
</div>
`;

exports[`AddFundsSourcePickerForUser component renders default options 1`] = `
<div>
  <select
    className="form-control"
    id="sourcePicker"
    name="template"
    onChange={[Function]}
    placeholder="select"
  >
    <option
      value=""
    />
  </select>
</div>
`;
